### PR TITLE
Add simple Bag of Words example, Fix Canned Estimator

### PR DESCRIPTION
**Pull request by MarkDaoust**
The main reason the canned Estimator was failing was that it was using the wrong kind of feature column. You were passing the word indexes as a numeric column... switching to:

    xc = tf.contrib.layers.sparse_column_with_integerized_feature('x',NUM_WORDS)
    xc = tf.contrib.layers.embedding_column(xc,30)

Mostly fixes it for me. There were a couple of other things I fixed:

  1) Use "post" in `sequence.pad_sequences`. The default adds padding to the start of the sequence, while the estimator expects you to pad the end.

  2) I pull out the sequence lengths since the Estimator does use them if supplied.

  3) I added the "bucket by sequence length" logic.

  4) I added some dropout ... but it still overfits (The DNN BOW could also use some dropout).

The simple Bag-of-Words example I also attached makes a nice performance baseline, but the first half of the notebook also explores how feature_columns work with these things.

The BOW example file also shows how to convert back from the keras word indices to text.

### CODE REVIEW

**Comment by hereismari** [comment]
Thanks so much for the contribution Mark, it looks great!
